docs/src/constraints.md

```diff
@@ -876,6 +876,60 @@ julia> @constraint(model, M * y + q ⟂ y)
 [y[1] + 2 y[2] + 5, 3 y[1] + 4 y[2] + 6, y[1], y[2]] ∈ MathOptInterface.Complements(2)
 ```
 
+## Special Ordered Sets (SOS1 and SOS2)
+
+A Special Ordered Set (SOS) is an ordered set of variables with particular characteristics.
+
+If a vector `x` is in a Special Ordered Set type 1 (SOS1), at most one element (or variable) of `x`
+can take a non-zero value, all others being zero, in a feasible solution. To induce ordering,
+`weights` can be provided; as such, they should be unique values. The `k`th element in the set
+corresponds to the `k`th weight in `weights`. 
+
+A SOS1 constraint is equivalent to:
+
+- `x[(i)] >= 0` for some `i`, where `(i)` is the induced ordering *i*th element
+- `x[(j)] == 0` for all `j != i`
+
+If a vector is in a Special Ordered Set type 2 (SOS2), no more than two adjacent elements may
+be non-zero in a feasibles solution. To induce ordering,
+`weights` can be provided; as such, they should be unique values. The `k`th element in the set
+corresponds to the `k`th weight in `weights`. 
```

Comment:
If a vector of variables `x` is in a Special Ordered Set of Type II (SOS2), then at most two elements can be non-zero, and if two elements are non-zero, they must be adjacent.

Because of the adjacency requirement, you should supply a weight vector (with unique elements) to induce an ordering of the variables. The `k`th element in the ordered set corresponds to the `k`th weight in `weights` when the weights are sorted.